### PR TITLE
Executable manager refactor

### PR DESCRIPTION
**Pull request by rottebds**

### CODE REVIEW

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/util/executable/ExecutableManager.groovy [hunk 1]
```diff
@@ -58,56 +62,31 @@ class ExecutableManager {
...
-
-        executableFile
-    }
-
-    String getPathOfExecutableFromRelativePath(String path, ExecutableType executableType) {
-        File executableFile = getExecutableFromRelativePath(path, executableType)
-
-        null == executableFile ? null : executableFile.absolutePath
+    String getExecutablePath(ExecutableType executableType, boolean searchSystemPath, String path) {
+        getExecutable(path, executableType, searchSystemPath)?.absolutePath
```
Comment (jrichardBD) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/util/executable/ExecutableManager.groovy]: Doesnt look like line 70 calls a method that exists

Comment (rottebds) [reply src/main/groovy/com/blackducksoftware/integration/hub/detect/util/executable/ExecutableManager.groovy]: Good catch, I forgot to refactor that when I changed the method signature.

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/CondaBomTool.groovy [hunk 2]
```diff
@@ -56,7 +56,7 @@ class CondaBomTool extends BomTool {
     public boolean isBomToolApplicable() {
         def containsFiles = detectFileManager.containsAllFiles(sourcePath, 'environment.yml')
         if (containsFiles) {
-            condaExecutablePath = executableManager.getPathOfExecutable(ExecutableType.CONDA, detectConfiguration.getCondaPath())
+            condaExecutablePath = getExecutablePath(ExecutableType.CONDA, true, detectConfiguration.getCondaPath())
```
Comment (jrichardBD) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/CondaBomTool.groovy]: Is this supposed to be `findExecutablePath(....)` not `getExecutablePath(...)` ?

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/CpanBomTool.groovy [hunk 3]
```diff
@@ -58,8 +58,8 @@ class CpanBomTool extends BomTool {
     public boolean isBomToolApplicable() {
         def containsFiles = detectFileManager.containsAllFiles(sourcePath, 'Makefile.PL')
         if (containsFiles) {
-            cpanExecutablePath = executableManager.getPathOfExecutable(ExecutableType.CPAN, detectConfiguration.getCpanPath())
-            cpanmExecutablePath = executableManager.getPathOfExecutable(ExecutableType.CPANM, detectConfiguration.getCpanmPath())
+            cpanExecutablePath = getExecutablePath(ExecutableType.CPAN, true, detectConfiguration.getCpanPath())
```
Comment (jrichardBD) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/CpanBomTool.groovy]: Is this supposed to be `findExecutablePath(....)` not `getExecutablePath(...)` ?

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/DockerBomTool.groovy [hunk 4]
```diff
@@ -62,8 +62,8 @@ class DockerBomTool extends BomTool {
         if (!propertiesOk) {
             logger.debug('The docker properties are not sufficient to run')
         } else {
-            dockerExecutablePath = executableManager.getPathOfExecutable(ExecutableType.DOCKER, detectConfiguration.dockerPath)?.trim()
-            bashExecutablePath = executableManager.getPathOfExecutable(ExecutableType.BASH, detectConfiguration.bashPath)?.trim()
+            dockerExecutablePath = getExecutablePath(ExecutableType.DOCKER, true, detectConfiguration.dockerPath)
```
Comment (jrichardBD) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/DockerBomTool.groovy]: Is this supposed to be `findExecutablePath(....)` not `getExecutablePath(...)` ?

**Comment by jrichardBD** [comment]
@rottebds I think you need to review the diff in this PR and correct the method calls

File: src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/nuget/NugetInspectorPackagerTest.groovy [hunk 5]
```diff
@@ -8,6 +9,7 @@ import com.blackducksoftware.integration.hub.detect.nameversion.NameVersionNodeT
 import com.google.gson.Gson
 import com.google.gson.GsonBuilder
```
Comment (jrichardBD) [inline src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/nuget/NugetInspectorPackagerTest.groovy]: Why are we ignoring this test?

Comment (JakeMathews) [reply src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/nuget/NugetInspectorPackagerTest.groovy]: Jordan's new version of the inspector broke all of his tests. They need to be rewritten. I don't think this change should be in this branch though.

**Pull request merged**